pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -423,6 +434,37 @@ func (c *Reconciler) reconcileMetric(ctx context.Context, pa *pav1alpha1.PodAuto
 	return nil
 }
 
+func (c *Reconciler) checkContainerHealth(ctx context.Context, pa *pav1alpha1.PodAutoscaler) error {
+	logger := logging.FromContext(ctx)
+
+	if pa.Status.IsActive() {
+		pa.Status.MarkContainerHealthy()
+	} else if pa.Status.IsActivating() {
+		deployment, err := c.deploymentLister.Deployments(pa.Namespace).Get(anames.Deployment(pa.Name))
+		if err != nil {
+			logger.Errorf("Error getting deployment: %v", err)
+			return err
+		}
+		pods, err := c.KubeClientSet.CoreV1().Pods(pa.Namespace).List(metav1.ListOptions{LabelSelector: metav1.FormatLabelSelector(deployment.Spec.Selector)})
+		if err != nil {
+			logger.Errorf("Error getting pods: %v", err)
+			return err
+		} else if len(pods.Items) > 0 {
+			// Arbitrarily grab the very first pod, as they all should be crashing if one does.
```

Comment:
I'm not sure if this holds -- trying to wrap my head around this:

In the revision reconciler, this was inside a check:
```go
// If a container keeps crashing (no active pods in the deployment although we want some)	
if *deployment.Spec.Replicas > 0 && deployment.Status.AvailableReplicas == 0 {
  ...
}
```

Does `pa.Status.IsActivating` imply that check is true?

Reply:
It was equivalent when I drafted this prototype, because pa.Status.Active becomes true as soon as one pod is available.  But now it is not, because pa.Status.Active only becomes true with a minimum number of replicas.  Brought that condition over.